### ISSUE TITLE
Bug: S1854: Linter shouldn't accept the variable used as the object key as dead storage.

### ISSUE THREAD

**Issue by vmikhav** [SonarSource/SonarTS#888]
**I want to report a bug.**

**SonarQube version:** 7.1 (build 11001) <br>

**Rule key:** typescript:S1854 (Dead stores should be removed)

**Reproducer** 
```TypeScript
// sound = 'meow' | 'woof'
const report = {'cat': 0, 'dog': 0};
let key = 'cat';
if (sound === 'woof') {
  key = 'dog';
}
report[key]++;
```

**Expected behavior**
Linter shouldn't accept the variable used as the object key as dead storage.

**Comment by saberduck** [comment]
hello @vmikhav , we are not maintaining this plugin anymore, as we migrated TypeScript analysis to [SonarJS plugin](https://github.com/SonarSource/SonarJS/). Can you try it to analyze with SonarJS? Though, you will need to upgrade SonarQube at least to the 7.9 LTS version.
